### PR TITLE
Problem Services

### LINKED ISSUE #184: Refactor Problem Services

### PR DESCRIPTION
**Pull request by carson-ricca**
## Description
Move Services into the Module

## Types of changes
What types of changes does your code introduce?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (only improve the code quality no change in functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] Issue has been created
- [x] Every file touched is linted
- [x] Every file touched has tests that cover all the funcitonality with green coverage
- [ ] Documentation is updated (if applicable).

## Issue
Please link the issue here.
Closes #184

## Tests
Please describe in detail what tests you added or changed.

### CODE REVIEW

**Comment by carson-ricca** [comment]
I have tested and the functionality works as expected. The services that are used in other models are in the `problems.module.ts` and work in the other modules they are used in. The services that are only used within the module are also working. Added some documentation to the services.

File: src/app/problems/problem-create/problem-create.component.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 import {ActivatedRoute} from '@angular/router';
 import {Subscription} from 'rxjs';
+import {QuestionService} from "@app/problems/_services/question.service";
 
 @Component({
     selector: 'app-problem-create',
     templateUrl: './problem-create.component.html',
-    styleUrls: ['./problem-create.component.scss']
+    styleUrls: ['./problem-create.component.scss'],
+    providers: [QuestionService]
```
Comment (keyvankhademi) [inline src/app/problems/problem-create/problem-create.component.ts:10]: WHy do we have the provider here?

File: src/app/problems/problems.module.ts [hunk 2]
```diff
@@ -68,6 +70,8 @@ import {DragulaModule} from "ng2-dragula";
         DragulaModule.forRoot(),
     ],
     providers: [
+        DifficultyService,
```
Comment (keyvankhademi) [inline src/app/problems/problems.module.ts:73]: Why didn't you add all the providers here?

File: src/app/problems/problem-create/problem-create.component.ts (hunk 1 above)
Comment (carson-ricca) [reply src/app/problems/problem-create/problem-create.component.ts:10]: The example I viewed for keeping a service within a module showcased it being done in this manner.

File: src/app/problems/problems.module.ts (hunk 2 above)
Comment (carson-ricca) [reply src/app/problems/problems.module.ts:73]: The stuff I searched up showed that you didn't need to add the services that are only used within the module to the providers.

**Review by keyvankhademi — APPROVED** [review]

**Pull request merged**